ocaml/typing/predef.ml

```diff
@@ -342,23 +342,35 @@ let build_initial_env add_type add_extension empty_env =
   |> add_type1 ident_iarray
        ~variance:Variance.covariant
        ~separability:Separability.Ind
+       ~param_jkind:(Jkind.Primitive.any_non_null ~why:Array_type_argument)
```

Comment:
This looks unrelated to this patch, but instead is a bugfix. Is that right? @dkalinichenko-js, does this look right to you?

Reply:
This allows `iarray`s to accept all layouts instead of just values. The nullability here is correct.